platform/kube/controller.go

```diff
@@ -416,8 +419,17 @@ func (c *Controller) GetIstioServiceAccounts(hostname string, ports []string) []
 // AppendServiceHandler implements a service catalog operation
 func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) error {
 	c.services.handler.Append(func(obj interface{}, event model.Event) error {
-		if svc := convertService(*obj.(*v1.Service), c.domainSuffix); svc != nil {
-			f(svc, event)
+		svc := *obj.(*v1.Service)
+
+		// Do not handle "kube-system" services
+		if svc.Namespace == meta_v1.NamespaceSystem {
```

Comment:
Please place a TODO to make this configurable. Eventually, istio will run in kube-system and will be subject to our control plane.